cli/boards.go

```diff
@@ -0,0 +1,234 @@
+package cli
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+	"go.uber.org/multierr"
+	packagepb "go.viam.com/api/app/packages/v1"
+)
+
+// supportedVersionRegex validates that the board version is semver 2.0.0 specification.
+var supportedVersionRegex = regexp.MustCompile(`^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)` +
+	`(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)` +
+	`(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`)
+
+const boardUploadChunkSize = 32 * 1024
+
+// UploadBoardDefsAction is the corresponding action for "board upload".
+func UploadBoardDefsAction(c *cli.Context) error {
+	orgArg := c.String(organizationFlag)
+	nameArg := c.String(boardFlagName)
+	versionArg := c.String(boardFlagVersion)
+	jsonPath := c.Args().First()
+	if c.Args().Len() > 1 {
```

Comment:
Mild preference for checking the length of the args before we call `First()` on them. and change this to making sure there's exactly 1 arg (because 0 is no good, and you already show that more than 1 is no good). 

...but if it's possible to bundle multiple files into a single package, multiple args here might be okay after all, so I'm unsure whether this is the right check.

Reply:
I think its good to have separate error messages depending on if they put too many arguments or don't put any, so I don't want to combine into just checking if its exactly 1.